ci/Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM registry.ddbuild.io/images/mirror/golang:1.19 AS build
```

Comment:
I would suggest making this closer to the official Dockerfile: https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol-contrib/Dockerfile, to avoid differences.

Some relevant differences:
- Official image is 'from scratch', while this one is based on xenial
- Official image runs as non-root
- Official image uses certificates from Alpine instead of Ubuntu (I doubt this makes a big difference, so I don't think this one is as important as the other two)

Reply:
If we're going to use their Dockerfile for inspiration, why don't we either:
1. Use the image it builds as a base image?
2. Fork it?

Reply:
Those could be good solutions, yes :) To me the important thing is that we are as close as possible to upstream, so that we are confident that we are testing the same thing.

Reply:
We should have both ubuntu images & scratch images. Ubuntu make debugging easy . I would add a new image `from scratch` in a follow up PR

Reply:
I don't think we need them if we use ephemeral containers https://kubernetes.io/docs/tasks/debug/debug-application/debug-running-pod/#ephemeral-container, but I guess if it's easy to maintain then that's okay with me